packages/react/src/analytics/integrations/Guardian/Guardian.ts

```diff
@@ -0,0 +1,146 @@
+/**
+ * Guardian Integration. Loads the Guardian script and calls its methods.
+ *
+ * @example <caption>Adding Guardian integration to analytics</caption>
+ * ```
+ *
+ * import \{ analytics, integrations \} from '\@farfetch/blackout-react/analytics';
+ *
+ * analytics.addIntegration('guardian', integrations.Guardian, \{\});
+ * ```
+ */
+
+import {
+  EventData,
+  integrations,
+  LoadIntegrationEventData,
+  SetUserEventData,
+  StrippedDownAnalytics,
+  TrackTypesValues,
+  utils,
+} from '@farfetch/blackout-analytics';
+import { Forter, Riskified } from './providers';
+import {
+  getEnvironmentFromOptions,
+  getSafeOptionsForIntegration,
+} from './guardian-helper';
+import {
+  type GuardianIntegrationBaseOptions,
+  type GuardianIntegrationOptions,
+  GuardianProviders,
+} from './types';
+import omit from 'lodash/omit';
+import type { ForterIntegrationOptions } from './providers/Forter/types';
+import type { RiskifiedIntegrationOptions } from './providers/Riskified/types';
+
+/**
+ * Guardian integration.
+ */
+class Guardian extends integrations.Integration<GuardianIntegrationOptions> {
+  forterInstance?: Forter;
+  riskifiedInstance?: Riskified;
+  baseOptions: GuardianIntegrationBaseOptions;
+  activeIntegrations: Array<string>;
+
+  /**
+   * Creates an instance of Guardian integration.
+   *
+   * @param options   - Integration options.
+   * @param loadData  - Analytics' load event data.
+   * @param analytics - Analytics stripped down instance.
+   */
+  constructor(
+    options: GuardianIntegrationOptions = {},
+    loadData: LoadIntegrationEventData,
+    analytics: StrippedDownAnalytics,
+  ) {
+    const baseOptions: GuardianIntegrationBaseOptions = omit(options, [
+      'riskified',
+      'forter',
+    ]);
+    super(options, loadData, analytics);
+
+    this.baseOptions = baseOptions;
+
+    baseOptions.environment = getEnvironmentFromOptions(options);
+
+    // load default if not set
+    this.activeIntegrations = baseOptions.activeIntegrations || [
+      GuardianProviders.Forter,
+      GuardianProviders.Riskified,
+    ];
+
+    // Check if activeIntegrations is an array
+    if (!Array.isArray(this.activeIntegrations)) {
+      utils.logger.error(
+        '[Analytics] Guardian - The value `activeIntegrations` from Guardian integration options must be an array.',
+      );
+      return;
```

Comment:
This should throw an error since the instance will not be in the correct state. This will make any call to `track`, `onSetUser` to fail later when the integration should be removed from the active integrations in analytics.

Reply:
Another question: Since the user has also to pass the `riskified` and `forter` options on the integration options for them to initialize is it necessary to have the `activeIntegrations` option as well? Can't we derive the active integrations from the `riskified` and `forter` options?

Reply:
Riskified and Forter do not currently require any parameters as they are all optional. The aim is to allow users to either use the default settings or customize them if needed.

By filling out the "Riskified" or "Forter" object in GuardianOptions, it may require additional code for tenants who only want to set up the default options.

This way, it becomes easy to understand that there is one property to customize the enabled integrations and any custom properties for each integration can be specified in their respective "sub-option object".